### ISSUE TITLE
Evaluate Front-end Frameworks for the site

### ISSUE THREAD

**Issue by tap87** [cul-it/mann-wagon#110]
Events prototype was done in Angular 1, mainly because I was familiar with it. We need to have evaluate and discuss which Framework to use for the site and potentially other project as well.

**Comment by tap87** [comment]
Nick and I discussed this today, and decided that we will take some time to compare a few JS frameworks (AngularJS 2, reactJS, ...), based on requirements for Mann website. For now, we will use AngulaJS 1 as that provides the fastest time of delivery.

**Comment by tap87** [comment]
Booked doesn't have an option to specify reservations types (workshops, classes, etc.). We can do that using custom attributes, but the api for getting all reservations doesn't include custom attributes. However, we can filter by a room(resource). Based on this information, Nick and I discussed a workaround for now.

- Tie an event type to a room in booked, stone class room for workshops only
- Get events from localist for 28 days (we will have ~40 events listed on April 21)
- Combine these two data sources, and display and filter results.

**Comment by cappadona** [comment]
Thanks @tap87. Just referencing #111 since this reflects on the capabilities of Booked even though this particular feature (type/categorized reservations) was missing from the requirements matrix :)

**Comment by tap87** [comment]
The team decided to evaluate three front-end libraries/frameworks, based on the following requirements.

1. Merge events data from two data sources (localist and booked api)
2. Render the events
3. Use a field (room location) to filter the events.

I found some similarities between Angular2, ReactJS and Vue.js in terms of having component oriented development. However, Angular2 and Vue.js are very similar, since Vue.js is based on Angular, and uses a lot of concepts from Angular 1 and 2.
Based on the requirements, we found creating filters much easier in Angular2, and Vue.js, as compared to ReactJS. ReactJS didn't have the concept of filters that Angular2 and Vue.js have.
It also seems that the trend in ReactJS is to pull everything in javascript including templates and css styles.
In terms of merging data from two sources, both Angular2 and Vue.js didn't offer any tools or libraries. For the initial iteration, I relied on jQuery to loop over the json objects. While Angular2 is a framework that comes with a lot of functionality, including routes, resources, etc, we felt that for this project it would complicate things too much in terms of sticking to AngularJS conventions, as compared to Vue.js. We also felt that the learning curve in Angular2 would be much higher compared to Vue.js.
Based on these factors, we decided to use Vue.js as the library for front-end development on the Mann  library website.
